Fix issue where a full url is quoted incorrectly

grantHarris opened the pull request:
This Pr fixes an issue reported where a full URL is quoted incorrectly. See https://github.com/marshmallow-code/marshmallow-jsonapi/issues/171

Commit:
Fix issue where a full url is quoted incorrectly

sloria commented:
Builds are now fixed on `dev`. @grantHarris Can you please update this with the `dev` branch?

grantHarris commented:
Merged dev in

sloria commented:
Passing in safe characters seems brittle. I'm wondering if we should go with @scottwernervt 's [suggestion](https://github.com/marshmallow-code/marshmallow-jsonapi/issues/171#issuecomment-432733049) and remove automatic quoting. The user can override generate_url to do any necessary quoting.

Thoughts?

grantHarris commented:
I did a bit more digging. You already have the werkzeug library included for flask support. We could use its url_fix function instead. See https://github.com/pallets/werkzeug/blob/master/werkzeug/urls.py#L592

Thoughts?

sloria commented:
Flask and werkzeug are not hard dependencies now. I'm not entirely opposed to adding werkzeug as a dependency, but I'm not sure it's necessary. In most cases, URLs won't include content from user input, right?

franklinchou commented:
@sloria Ideally, yes. URLs should be generated by the client not from user input. In that case I don't see any reason not to get this merged in.

sloria commented:
The alternative suggested above is to remove the `quote` call altogether. @grantHarris Are you good with that approach?

sloria committed:
Remove quoting

As suggested in https://github.com/marshmallow-code/marshmallow-jsonapi/issues/171#issuecomment-432733049,
quoting can be achieved by overriding generate_url.

sloria commented:
I've gone ahead and removed the quoting entirely. As suggested in https://github.com/marshmallow-code/marshmallow-jsonapi/issues/171#issuecomment-432733049, users can override `generate_url` if they quoting.
